src/vs/workbench/parts/debug/node/debugAdapter.ts

```diff
@@ -28,34 +29,74 @@ export class Adapter {
 		}
 	}
 
-	public get runtime(): string {
+	public getAdapterExecutable(verifyAgainstFS = true): TPromise<IAdapterExecutable> {
+
+		if (this.rawAdapter.adapterExecutableCommand) {
+			return this.commandService.executeCommand<IAdapterExecutable>(this.rawAdapter.adapterExecutableCommand).then(ad => {
+				return this.verifyAdapterDetails(ad, verifyAgainstFS);
+			});
+		}
+
+		const ad = <IAdapterExecutable>{
+			command: this.getProgram(),
+			args: this.getAttributeBasedOnPlatform('args')
+		};
+		const runtime = this.getRuntime();
+		if (runtime) {
+			const runtimeArgs = this.getAttributeBasedOnPlatform('runtimeArgs');
+			ad.args = (runtimeArgs || []).concat([ad.command]).concat(ad.args || []);
+			ad.command = runtime;
+		}
+		return this.verifyAdapterDetails(ad, verifyAgainstFS);
+	}
+
```

Comment:
The purpose of `verifyAdapterDetails` is to verify the IAdapterExecutable structure. The current implementation verifies basically two things:
- if command is not empty
- if the command exists in the file system 

I had to introduce `verifyAgainstFS` because the tests are using paths that do not exist on disk, so the test must be able to disable the second part of the verification. The first part is still done.